account_id.go

```diff
@@ -112,7 +124,12 @@ func (id AccountID) ToStringWithChecksum(client *Client) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return fmt.Sprintf("%d.%d.%d-%s", id.Shard, id.Realm, id.Account, checksum.correctChecksum), nil
+
+	if id.AliasKey == nil {
+		return fmt.Sprintf("%d.%d.%d-%s", id.Shard, id.Realm, id.Account, checksum.correctChecksum), nil
+	} else {
+		return fmt.Sprintf("%d.%d.%s-%s", id.Shard, id.Realm, id.AliasKey.String(), checksum.correctChecksum), nil
```

Comment:
We should error here since `checksum` is not supported by `AccountID` with an `AliasKey`